### PR TITLE
Initialize Data Correctly For Powchain Service

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] Handles empty trie with 0 elements in it correctly.
- [x] Abstract method into separate functions.
- [x] Adds in unit tests

**Which issues(s) does this PR fix?**

**Other notes for review**

### CODE REVIEW

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -936,6 +925,52 @@ func (s *Service) fallbackToNextEndpoint() {
...
+	// Exit for empty containers.
+	if ctrLen == 0 {
+		return true
+	}
+	for _, c := range ctrs {
+		if c.Index == 0 {
+			return true
+		}
+	}
+	log.Infof("Recovering missing deposit containers, node is re-requesting missing deposit data.")
```
Comment (prestonvanloon) [inline beacon-chain/powchain/service.go:970]:
```suggestion
	log.Info("Recovering missing deposit containers, node is re-requesting missing deposit data.")
```

Comment (terencechain) [reply beacon-chain/powchain/service.go:970]: Remove the period too. Thanks

File: beacon-chain/powchain/service.go [hunk 2]
```diff
@@ -936,6 +925,53 @@ func (s *Service) fallbackToNextEndpoint() {
...
+	if len(items) == 1 && bytes.Equal(items[0], params.BeaconConfig().ZeroHash[:]) {
+		s.lastReceivedMerkleIndex = -1
+	}
+	if err := s.initDepositCaches(ctx, eth1DataInDB.DepositContainers); err != nil {
+		return errors.Wrap(err, "could not initialize caches")
+	}
+	return nil
+}
+
+// validates that all deposit containers are valid and have their relevant indexes
```
Comment (rauljordan) [inline beacon-chain/powchain/service.go:958]:
```suggestion
// validates that all deposit containers are valid and have their relevant indices
```

**Review by terencechain — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

File: beacon-chain/powchain/service.go [hunk 3]
```diff
@@ -955,7 +955,7 @@ func (s *Service) initializeEth1Data(ctx context.Context, eth1DataInDB *protodb.
 	return nil
 }
 
-// validates that all deposit containers are valid and have their relevant indexes
+// validates that all deposit containers are valid and have their relevant indices
```
Comment (dustpan28646) [inline beacon-chain/powchain/service.go:958]: how this work

**Pull request merged**